version: "3.8"

networks:
  toolshare_network:

services:
  db:
    image: postgres:alpine
    container_name: postgres_container
    environment:
      POSTGRES_USER: myuser
      POSTGRES_PASSWORD: password
      POSTGRES_DB: fastapi_database
    volumes:
      - postgres_data:/var/lib/postgresql/data
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql  # Assumes init.sql creates the test database
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "myuser"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - toolshare_network

  web:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: fastapi_app
    env_file:
    - .env.production
    ports:
      - "8000:8000"
    depends_on:
      db:
        condition: service_healthy
    volumes:
    - ./app.log:/app/app.log
    networks:
      - toolshare_network


  test:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: fastapi_tests
    env_file:
      - .env.testing
    depends_on:
      db:
        condition: service_healthy
    command: ["pytest", "-v", "--ignore=WithoutSecurity"]
    networks:
      - toolshare_network

volumes:
  postgres_data:
